lib/service/cfg.go

```diff
@@ -301,7 +301,9 @@ func (cfg *Config) ApplyCAPins(caPins []string) error {
 		}
 		filteredPins = append(filteredPins, strings.Split(pins, "\n")...)
 	}
-	cfg.CAPins = filteredPins
+	if len(filteredPins) > 0 {
+		cfg.CAPins = filteredPins
```

Comment:
I'm actually curious, why do we override them from the CLI flags? Wouldn't appending make more sense?

Reply:
That was the prior behavior. It's seemingly the way "merging" cli/file config is done for everything - cli overrides file config settings. But since you mention it, I guess we could do that? The only other slice type CLI flag I see in `teleport` is `--auth-server` which also overrides the file config.

Reply:
I'm also curious why empty ca pin is just a warning? It's easy to miss the warning in the log, especially with --debug. Why not gate it behind `--insecure` or a new flag `--insecure-no-ca-pin`?

Reply:
> I'm also curious why empty ca pin is just a warning? It's easy to miss the warning in the log, especially with --debug. Why not gate it behind --insecure or a new flag --insecure-no-ca-pin for first connect?

_In theory_ you don't need a CA pin when joining in IoT mode because the proxy cert should be verified by PKI, however, there's https://github.com/gravitational/teleport-private/issues/135. The reason for direct joining to auth is backward compatibility. I've proposed similar flags, and when I (or anyone) get time to work on it we'll be more strict about this